### PR TITLE
chore: taking db dependencies out of requirements-dev.txt

### PR DESCRIPTION
**Pull request by mistercrunch**
### CATEGORY
- [x] Build / Development Environment

### SUMMARY
Py deps `mysqlclient` and `psycopg2` really aren't dev dependencies, they're more optional deps, so the right place for them is `extra_requires` in our `setup.py`.

Not a big deal, but `mysqlclient` has os-level deps and that forces people setting up a dev env to jump through unnecessary hoops

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -112,9 +112,12 @@ def get_git_sha():
         'hive': [
             'pyhive[hive]>=0.6.1',
             'tableschema',
+            'thrift>=0.11.0, <1.0.0',
         ],
         'presto': ['pyhive[presto]>=0.4.0'],
         'gsheets': ['gsheetsdb>=0.1.9'],
+        'mysql': ['mysqlclient==1.3.13'],
```
Comment (john-bodley) [inline setup.py]: Any reason to pin these here? Note we may need to have a `requirements.in` file including the line`-e .` (and possibly mention of the extra targets) is we want to have a fully pinned environment for testing purposes.

File: tox.ini [hunk 2]
```diff
@@ -63,6 +63,18 @@ setenv =
 whitelist_externals =
     npm
 
+[testenv:py36-mysql]
+deps =
+    -rrequirements.txt
+    -rrequirements-dev.txt
+    .[mysql]
```
Comment (john-bodley) [inline tox.ini]: See my comment regarding a `requirements.in` file which would remove the need for this.

File: setup.py (hunk 1 above)
Comment (mistercrunch) [reply setup.py]: `pip-compile` doesn't play nicely with extras, there's no way to say give me the descendants on [only] the extras. Related comment in a related PR here: https://github.com/apache/incubator-superset/pull/7372#issuecomment-494243758

File: setup.py [hunk 3]
```diff
@@ -112,9 +112,12 @@ def get_git_sha():
         'hive': [
             'pyhive[hive]>=0.6.1',
             'tableschema',
+            'thrift>=0.11.0, <1.0.0',
         ],
         'presto': ['pyhive[presto]>=0.4.0'],
         'gsheets': ['gsheetsdb>=0.1.9'],
+        'mysql': ['mysqlclient==1.4.2.post1'],
```
Comment (john-bodley) [inline setup.py]: Nit. Alphabetical ordering.

**Review by john-bodley — APPROVED** [review]

**Pull request merged**